source/parse.h

```diff
@@ -8506,7 +8507,10 @@ class parser
                 //  Then there may be a semicolon
                 //  If there is a semicolon, eat it
                 if (!done() && curr().type() == lexeme::Semicolon) {
-                    next();
+                    if (semicolon_allowed)
+                      next();
```

Comment:
I'll tweak this to match house style == braces around all branch bodies, and 4-space indent.